runtime/lua/vim/lsp.lua

```diff
@@ -1220,6 +1220,51 @@ function lsp.buf_attach_client(bufnr, client_id)
   return true
 end
 
+--- Detaches client from the specified buffer.
+--- Note: While the server is notified that the text document (buffer)
+--- was closed, it is still able to send notifications should it ignore this notification.
+---
+---@param bufnr number Buffer handle, or 0 for current
+---@param client_id number Client id
+function lsp.buf_detach_client(bufnr, client_id)
+  validate {
+    bufnr     = {bufnr, 'n', true};
+    client_id = {client_id, 'n'};
+  }
+  bufnr = resolve_bufnr(bufnr)
+
+  local client = lsp.get_client_by_id(client_id)
+  if not client or not client.attached_buffers[bufnr] then
+    vim.notify(
+      string.format('Buffer (id: %d) is not attached to client (id: %d). Cannot detach.', client_id, bufnr)
+    )
+    return
+  end
+
+  changetracking.reset_buf(client, bufnr)
+
+  local uri = vim.uri_from_bufnr(bufnr)
+  local params = { textDocument = { uri = uri; } }
+
+  if client.resolved_capabilities.text_document_open_close then
+    client.notify('textDocument/didClose', params)
```

Comment:
```suggestion
  if client.resolved_capabilities.text_document_open_close then
    local uri = vim.uri_from_bufnr(bufnr)
    local params = { textDocument = { uri = uri; } }
    client.notify('textDocument/didClose', params)
```

Looks like `params` are otherwise not used.